### PR TITLE
fix: replace `DEVELOP` mode

### PR DESCRIPTION
**Pull request by simonrw**
## Motivation

During the migration to the new runtime ([PR 1 adding the new runtime](https://github.com/localstack/localstack/pull/10942), [PR 2 removing the old runtime](https://github.com/localstack/localstack/pull/11202)), the `DEVELOP` live debugging functionality was missed. This workflow, coupled with using `localstack.dev.run` is crucial for debugging parts of LocalStack that _only_ work when run from a container, e.g. testing inter-container Docker networking via services.

## Changes

- Add the previous `DEVELOP` functionality back, including launching the `debugpy` debugger
  - enable the feature with `DEVELOP=1`
  - pause the LocalStack start on debugger connection with `WAIT_FOR_DEBUGGER=1`
  - configure the port used by `debugpy` with `DEVELOP_PORT` (default: 5678)
- Fix a bug in `localstack.dev.run` where community only entrypoints were not being mounted from the correct place (when launched from a community checkout)
- Fix a bug in `localstack.dev.run` where community and pro entrypoints were not being mounted from the correct place (when launched from a pro checkout)
- Add convenience properties on `HostPaths` to access source code checkout paths on the host

## Testing

* checkout `master`
* run `localstack.dev.run --mount-entrypoints`
    * notice that the community entrypoints file is not mounted into the container
* checkout this PR
* run `localstack.dev.run --mount-entrypoints`
    * notice that the community entrypoints file **is** mounted into the container
* run `localstack.dev.run --mount-entrypoints --develop`
    * notice that the logs indicate that `debugpy` is being installed, and is available on port 5678

### CODE REVIEW

File: localstack-core/localstack/debugger/plugins.py
Comment (thrau) [inline localstack-core/localstack/debugger/plugins.py:1]: how about we move this into `localstack.dev` ?

File: localstack-core/localstack/debugger/plugins.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    import debugpy  # noqa: T100
+
+    LOG.info("Starting debug server at: %s:%s", constants.BIND_HOST, config.DEVELOP_PORT)
+    debugpy.listen((constants.BIND_HOST, config.DEVELOP_PORT))  # noqa: T100
+
+    if config.WAIT_FOR_DEBUGGER:
+        debugpy.wait_for_client()  # noqa: T100
+
+
+@hooks.on_infra_start()
```
[the remark this answers is not in the document]
Comment (thrau) [reply localstack-core/localstack/debugger/plugins.py:22]: depends - are you looking into also debugging the startup procedure? in which case we'd probably need to think the whole approach.

**Review by thrau — APPROVED** [review]
LGTM! just one comment related to the code organization

File: localstack-core/localstack/debugger/plugins.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonrw) [reply localstack-core/localstack/debugger/plugins.py:22]: We can leave it unspecified for now and adjust if someone (likely me by the sounds of it) asks for it

**Pull request merged**